### PR TITLE
U/mattc/deb

### LINKED ISSUE #30: Create a .deb for Ubuntu
There is no way in which pidtree-bcc *needs* to run in a container.

Create a parallel packaging flow that creates a debian package for Ubuntu (which will hopefully Just Work on Debian too).

It will notably *not* include:
- A systemd unit file, as this is currently the control plane for directing multi-tenant instances to their respective log streams and dependent services, which is business logic dependent
- Any explicit dependencies on the linux kernel headers as there is no way (as far as I am aware) to create a depency on the correct version (e.g. installing `linux-headers-generic` will install the headers for the latest version, not the one you're running)

### PR DESCRIPTION
**Pull request by oholiab**
Fixes #30 

Notably there's an environment variable to set for installing the headers at runtime - it does not do this by default.

I'll follow up with documentation once it's merged just in case there are concerns about my methodology here. It'll need all the caveats of the runtime dependencies.

### CODE REVIEW

**Comment by oholiab** [comment]
Breakages appear to mostly be `make` being a dork.

**Comment by oholiab** [comment]
Okay I'm not entirely sure why it's being an ass this time, will fix it up tomorrow but I can assure you that the itests do in fact pass on a real computer, so review as if they have and I'll get it working again.

File: itest/itest.sh [hunk 1]
```diff
@@ -63,18 +64,32 @@ function wait_for_tame_output {
...
-  # Run the setup.sh install steps in the image so we don't hit timeouts
-  docker build -t pidtree-itest itest/
   mkfifo $FIFO_NAME
-  echo "Launching itest-container $CONTAINER_NAME"
-  docker run --name $CONTAINER_NAME -d\
-      --rm --privileged --cap-add sys_admin --pid host \
-      -v $(git rev-parse --show-toplevel)/itest/example_config.yml:/work/config.yml \
-      -v $(git rev-parse --show-toplevel)/$FIFO_NAME:/work/outfile \
-      pidtree-itest -c /work/config.yml -f /work/outfile
+  if [[ "$@" = "" ]]; then
```
Comment (Qmando) [inline itest/itest.sh]: If i type `make itest_xenial` this `$@` variable becomes "--xenial", so neither a docker or debian is created and the tests fail. Perhaps we should catch that situation and give the user back an error about what make itest_ targets are supported

Comment (oholiab) [reply itest/itest.sh]: ++, fixed

**Pull request merged**